File: src/main/java/it/contrader/service/CategoryService.java
package it.contrader.service;

import it.contrader.converter.CategoryConverter;
import it.contrader.dao.CategoryDAO;
import it.contrader.dto.CategoryDTO;
import it.contrader.model.Category;

public class CategoryService extends AbstractService<Category, CategoryDTO> {

	public CategoryService() {
		this.dao = new CategoryDAO();
		this.converter = new CategoryConverter();
	}
}

File: src/service/category.service.ts
import { Injectable } from '@angular/core';
import { AbstractService } from './abstractservice';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CategoryDTO } from 'src/dto/categorydto';

@Injectable({
  providedIn: 'root'
})
export class CategoryService extends AbstractService<CategoryDTO> {

  constructor(http: HttpClient) {
    super(http);
    this.type = 'category';
   }
}
